### ISSUE TITLE
Multiple foreign keys result in SQL Error

### ISSUE THREAD

**Issue by fridolin-koch** [strongloop/loopback-connector-mysql#254]
# Description/Steps to reproduce

1. Create a model with multiple foreignKeys:

```json
"options": {
    "mysql": {
      "table": "deliveries"
    },
    "foreignKeys": {
      "fk_deliveries_stateId": {
        "name": "fk_deliveries_stateId",
        "entity": "DeliveryState",
        "entityKey": "id",
        "foreignKey": "stateId"
      },
      "fk_deliveries_orderId": {
        "name": "fk_deliveries_orderId",
        "entity": "Order",
        "entityKey": "id",
        "foreignKey": "orderId"
      }
    }
  },
```

2. Exec Autoupdate

```javascript
app.datasources.mysql..autoupdate(null, function(err) {
  if (err) throw err;
  console.log('DONE');
  ds.disconnect();
});
```

# Expected result

Valid SQL is generated and executed.

Error Message:

```
Error: ER_PARSE_ERROR: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'ADD  CONSTRAINT `fk_parcels_carrierId` FOREIGN KEY (carrierId) REFERENCES `carri' at line 1
    at Query.Sequence._packetToError (/path/to/project/node_modules/mysql/lib/protocol/sequences/Sequence.js:52:14)
    at Query.ErrorPacket (/path/to/project/node_modules/mysql/lib/protocol/sequences/Query.js:77:18)
    at Protocol._parsePacket (/path/to/project/node_modules/mysql/lib/protocol/Protocol.js:280:23)
    at Parser.write (/path/to/project/node_modules/mysql/lib/protocol/Parser.js:75:12)
    at Protocol.write (/path/to/project/node_modules/mysql/lib/protocol/Protocol.js:39:16)
    at Socket.<anonymous> (/path/to/project/node_modules/mysql/lib/Connection.js:103:28)
    at emitOne (events.js:96:13)
    at Socket.emit (events.js:191:7)
    at readableAddChunk (_stream_readable.js:178:18)
    at Socket.Readable.push (_stream_readable.js:136:10)
    at TCP.onread (net.js:563:20)
    --------------------
    at Protocol._enqueue (/path/to/project/node_modules/mysql/lib/protocol/Protocol.js:141:48)
    at PoolConnection.query (/path/to/project/node_modules/mysql/lib/Connection.js:208:25)
    at runQuery (/path/to/project/node_modules/loopback-connector-mysql/lib/mysql.js:186:16)
    at executeWithConnection (/path/to/project/node_modules/loopback-connector-mysql/lib/mysql.js:228:7)
    at Ping.onOperationComplete [as _callback] (/path/to/project/node_modules/mysql/lib/Pool.js:110:5)
    at Ping.Sequence.end (/path/to/project/node_modules/mysql/lib/protocol/sequences/Sequence.js:86:24)
    at Ping.Sequence.OkPacket (/path/to/project/node_modules/mysql/lib/protocol/sequences/Sequence.js:95:8)
    at Protocol._parsePacket (/path/to/project/node_modules/mysql/lib/protocol/Protocol.js:280:23)
    at Parser.write (/path/to/project/node_modules/mysql/lib/protocol/Parser.js:75:12)
    at Protocol.write (/path/to/project/node_modules/mysql/lib/protocol/Protocol.js:39:16)
    at Socket.<anonymous> (/path/to/project/node_modules/mysql/lib/Connection.js:103:28)
    at emitOne (events.js:96:13)
    at Socket.emit (events.js:191:7)
    at readableAddChunk (_stream_readable.js:178:18)
    at Socket.Readable.push (_stream_readable.js:136:10)
    at TCP.onread (net.js:563:20)
```

I've created a fork and fixed the error, but I can't judge if this leads to any side effects: 

https://github.com/fridolin-koch/loopback-connector-mysql/commit/2b3edec386b5488090344d3edbd1abe2a2f9c6b7

# Additional information

OS: `darwin x64 7.7.4`
MySQL-Version: `5.7.17`

```
├── loopback@3.5.0
├── loopback-boot@2.23.0
├── loopback-component-explorer@4.2.0
├── loopback-component-model-diagram@0.3.0 (git+https://github.com/dmtw/loopback-component-model-diagram.git#4edf1f66a3b6b4ca5fb0fb0ec45a590f4ef8646a)
├── loopback-component-model-extender@0.1.4 (git+https://github.com/fridolin-koch/loopback-component-model-extender.git#08cd9c1e3e0fe5cffde0bbabaeffa6b765d64b7c)
├── loopback-connector-mysql@3.0.0 (git+https://github.com/strongloop/loopback-connector-mysql.git#3b2854045a80bd46c244c65cd0bc68ef61386b81)
├── loopback-ds-timestamp-mixin@3.4.1
├── loopback-setup-remote-methods-mixin@1.0.3
```

If this is not the expected behaviour I can send a PR

Best Regards,
Frido

**Comment by dhmlau** [comment]
@fridolin-koch , since you have a proposed fix, could you please submit a PR?  And please add related test cases as well.  Thanks!

**Comment by dhmlau** [comment]
@fridolin-koch, could you please create a PR?  We can go from there...   Thanks.

**Comment by darknos** [comment]
@dhmlau, @fridolin-koch I just committed PR with the fix #272

**Comment by ssh24** [comment]
Issue should be fixed in https://github.com/strongloop/loopback-connector-mysql/pull/272. Closing this is as a result.
